src/sensor/ping.cpp

```diff
@@ -72,29 +71,26 @@ Ping::Ping() : Sensor()
 
     //connectLink(LinkType::Serial, {"/dev/ttyUSB2", "115200"});
 
-    connect(&_detector, &ProtocolDetector::connectionDetected, this, &Ping::connectLink);
-    _detector.start();
+    connect(_detector, &ProtocolDetector::connectionDetected, this, [this] {
```

Comment:
where is connectLink() called after the detector finishes?

Reply:
the connection is done by sensor class, Ping (as a device class) should not deal with such things, only with things that are for ping device. Besides that, the protocol detection function is something that all devices will probably have, that's why I move it to sensor.